src/inc/til/generational.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    struct generation_t
+    {
+        auto operator<=>(const generation_t&) const = default;
+
+        constexpr void bump() noexcept
+        {
+            _value++;
+        }
+
+        uint32_t _value = 0;
+    };
+
+    template<typename T>
+    struct generational
+    {
+        generational() = default;
+        explicit constexpr generational(auto&&... args) :
+            _value{ std::forward<decltype(args)>(args)... } {}
+        explicit constexpr generational(generation_t generation, auto&&... args) :

```

Comment:
On the first frame we render we should ensure that all state is marked as invalidated, even if that state was never changed by the user. For instance, if the DPI remains at the default 96 we should still mark the DPI as having changed, so that the renderer will read it.

For instance in the future, we could create the IRenderData struct with generationals that all have an initial generation of 1, whereas the renderers will get a default-constructed one with a generation 0.